### PR TITLE
release: Update everything in this repo related to the release and its process

### PR DESCRIPTION
**Pull request by fidencio**
This commit is the first needed part in order to solve https://github.com/kata-containers/kata-containers/issues/9064, which will get a final decision on Feb 27, in the Kata Containers AC meeting.

For more information, please, see each commit message.

I'm leaving this as a draft till we get the official AC confirmation.

### CODE REVIEW

File: tools/packaging/release/release.sh
Comment (stevenhorsman) [inline tools/packaging/release/release.sh:155]: Should this be `$RELEASE_VERSION` rather than `$RELEASE`?

File: tools/packaging/release/release.sh
Comment (stevenhorsman) [inline tools/packaging/release/release.sh:177]: `$RELEASE_VERSION` instead of `$RELEASE_VERSIOB`

File: tools/packaging/release/release.sh
Comment (stevenhorsman) [inline tools/packaging/release/release.sh:171]: `${repo_url}` instead of `$repo_url}`

File: tools/packaging/release/release.sh
Comment (stevenhorsman) [inline tools/packaging/release/release.sh:177]: Optional suggestion: I personally prefer variables to have braces to avoid a brace expansion bear trap I've fallen into in the past and I find it clearer, as `${tarball}.asc` instead of `$tarball.asc`, but that might just be me.

File: tools/packaging/release/release.sh
Comment (stevenhorsman) [inline tools/packaging/release/release.sh:105]: I'm not sure if you want the `[ -z "$RELEASE_VERSION" ] && RELEASE_VERSION=$(cat "$repo_root_dir/VERSION")` code in this function for consistency with the others. Or maybe just extract that and run it at the start of the file outside of functions?

File: tools/packaging/release/release.sh
Comment (stevenhorsman) [inline tools/packaging/release/release.sh:58]: I think there is a bug in here that a double digit release with a minor update e.g. 3.10.0 will be stripped and end up as 3.2.0? and I assume we'd want to support this with monthly minor releases?

**Review by stevenhorsman — COMMENTED** [review]
I know this is just a draft, so I hope review comments are welcome still. I've had a look through and have a few suggestions. Thanks though it's generally looking good

File: tools/packaging/release/release.sh
Comment (stevenhorsman) [reply tools/packaging/release/release.sh:132]: Yeah - next release is set at the start of the workflow, so this isn't an issue

File: tools/packaging/release/release.sh
Comment (fidencio) [reply tools/packaging/release/release.sh:108]: Exactly!

File: tools/packaging/release/release.sh
Comment (fidencio) [reply tools/packaging/release/release.sh:177]: Oh, thanks for catching this one!

File: tools/packaging/release/release.sh
Comment (fidencio) [reply tools/packaging/release/release.sh:58]: Oh, that's a bug indeed! :-)

File: tools/packaging/release/release.sh
Comment (fidencio) [reply tools/packaging/release/release.sh:41]: It should, thanks!

File: tools/packaging/release/release.sh
Comment (fidencio) [reply tools/packaging/release/release.sh:105]: Not for create release, as right now we're passing that env var, and I'm not sure what would be state of the repo on the next job of the same workflow after pushing the new version file.

Comment (stevenhorsman) [reply tools/packaging/release/release.sh:105]: Yep - that's fair. After I reviewed this commit I saw the workflow with the `with`, but I thought it might be worth leaving the discussion about whether a refactor might be helpful

File: tools/packaging/release/release.sh
Comment (stevenhorsman) [reply tools/packaging/release/release.sh]: Looks good to me now. Thanks!

File: .github/workflows/release.yaml
Comment (stevenhorsman) [reply .github/workflows/release.yaml:97]: Yeah - it looks like it's only referenced in release.yaml, so one of us should try and have a mental todo to remove it after this gets merged.

**Review by stevenhorsman — APPROVED** [review]
Thanks for the fixes and improvements. LGTM (pending committee agreement to switch to this release procedure)!

File: docs/Release-Process.md
Comment (fidencio) [reply docs/Release-Process.md]: Just updated the docs, thanks!

**Review by gkurz — COMMENTED** [review]
Hey @fidencio ! Impressive work and what a magnificient diffstat ! :wink:

Left a few comments, one of which may address a bug.

I am impatient we switch to this :smile:

File: docs/Release-Process.md
Comment (fidencio) [reply docs/Release-Process.md:41]: I've changed the formatting of the whole document, to be honest.
I understand this is not optimal, but it gets easier and I think it's worth it due to how used this document is.

File: .github/workflows/release.yaml
Comment (fidencio) [reply .github/workflows/release.yaml]: Fixed, thanks!

File: tools/packaging/release/release.sh
Comment (fidencio) [reply tools/packaging/release/release.sh]: Fixed, thanks!

File: tools/packaging/release/release.sh
Comment (fidencio) [reply tools/packaging/release/release.sh]: Adopted the suggestion, thanks!

File: docs/Release-Process.md
Comment (gkurz) [inline docs/Release-Process.md:18]:
```suggestion
Those actions are manually triggered and are responsible for generating a new
```

**Comment by fidencio** [comment]
I've moved this out of the draft state to the "Ready for review" state as we have a 7/7 agreement on the AC about this.

**Pull request merged**
